sqlalchemy_cockroachdb/ddl_compiler.py

```diff
@@ -14,3 +28,166 @@ def visit_computed_column(self, generated, **kw):
         return "AS (%s) STORED" % self.sql_compiler.process(
             generated.sqltext, include_table=False, literal_binds=True
         )
+
+
+# TODO: convert visitors to memeber functions on CockroachDDLCompiler (it's like
+# this now just because I wrote+tested it inside our codebase).
+
+
+@compiles(CreateTable, "cockroachdb")
+def visit_create_table(
+    element: CreateTable, compiler: CockroachDDLCompiler, **kw: Any
+) -> str:
+    out = compiler.visit_create_table(element, **kw)
+
+    assert isinstance(element.target, Table)
+
+    if len(element.target.indexes) > 0:
+        indexes = [
+            _codegen_index(i, compiler, include_schema=False, **kw)
+            for i in element.target.indexes
+        ]
+
+        # TODO: Not compatible with anything that uses post_create_table, we
+        # need to parse properly to find the `)` which matches `CREATE TABLE (`.
+        out = out.rstrip().rstrip(")").rstrip()
+        out += ",\n"
+        out += ",\n\t".join(indexes)
+        out += "\n)"
+
+        # Record that we created these indexes so that we can double check it
+        # later.
+        for index in element.target.indexes:
+            index.info["_cockroachdb_index_created_by_create_table"] = True
+
+    return out
+
+
+@compiles(CreateIndex, "cockroachdb")
+def visit_create_index(element: Any, compiler: CockroachDDLCompiler, **kw: Any) -> str:
+    index = element.target
+    assert isinstance(index, Index)
+    was_created = index.info.get("_cockroachdb_index_created_by_create_table", False)
+    assert was_created
+
+    return "SELECT 'No-op: in cockroachdb we put index creation DDL inside the corresponding CREATE TABLE for improved performance.'"
+
+
+# Copy+paste of private function DDLCompiler._prepared_index_name
+def _prepared_index_name(
+    index: Index, compiler: DDLCompiler, include_schema: bool = False
+) -> str:
+    if index.table is not None:
+        effective_schema = compiler.preparer.schema_for_object(index.table)
+    else:
+        effective_schema = None
+    if include_schema and effective_schema:
+        schema_name = compiler.preparer.quote_schema(effective_schema)
+    else:
+        schema_name = None
+
+    index_name: str = cast(str, compiler.preparer.format_index(index))
+
+    if schema_name:
+        index_name = schema_name + "." + index_name
+    return index_name
+
+
+IDX_USING = re.compile(r"^(?:btree|hash|gist|gin|[\w_]+)$", re.I)
+
+
+# Heavily based on DDLCompiler.visit_create_index
```

Comment:
This is almost PostgresqlDDLCompiler.visit_create_index. Differences are:

* Remove the `CREATE` and `ON {table_name}` bits
* Replacing `USING` with `INVERTED` (seems to be needed for crdb?)
* Removing/commenting some features that I don't think crdb supports.

In the final version I would clean this up.

I don't think we should attempt to reuse `PostgresqlDDLCompiler.visit_create_index` - I think the string munging required for that would be quite brittle.